### DISCUSSION TITLE
Development Issue

### DISCUSSION THREAD

**Discussion by graciepii** [casact/chainladder-python#514] [Q&A]
Hello, I am currently using the below line of code, which works completely as expected for grain OYDY triangles. 

dev = cl.Development(n_periods=-1, drop_high=False, drop_low=False).fit_transform(tri)

However when I use OYDM it displays the below error:

ValueError: all keys need to be the same shape

Could anyone help?

**Comment by kennethshsu** [comment]
Hi @graciepii, welcome! Do you happen to have a sample dataset already? I might be able to help.

**Reply by graciepii** [reply]
Hi Kenneth, thanks for the reply! Unfortunately I can't provide the dataset as it is company data, and annoyingly have been unable to replicate the issue with sample data. 

It is extremely strange to me that dev = cl.Development().fit_transform(tri) works for the data at M or Q grain but the above does not. The issue does seem to be located in the handling of drop_high and _low specifically, and looking through the library it could be due to the way it sets up 3D arrays for these parameters in line 484 of base.

Any pointers on why this issue could be occurring would be greatly appreciated, even if it is not an exact answer!

**Comment by kennethshsu** [comment]
Ok, not a problem, I was able to reproduce the issue.

```python
import chainladder as cl
tri = cl.load_sample("prism")["Paid"].sum().grain("OYDQ")
cl.Development(drop_high = False).fit_transform(tri)
```

This is a bug.

**Reply by graciepii** [reply]
Thanks for confirming, is there any workaround to this at all? As development patterns on a yearly grain are not that helpful for short tailed classes.

**Reply by kennethshsu** [reply]
Hmm, nothing comes to mind. I'll try to fix this soon.

**Comment by kennethshsu** [comment]
@graciepii this should be resolved now!
